app/scripts/controllers/threebox.js

```diff
@@ -0,0 +1,113 @@
+const ObservableStore = require('obs-store')
+const Box = require('3box')
+
+class ThreeBoxController {
+  constructor (opts = {}) {
+    const { preferencesController, keyringController, addressBookController, provider } = opts
+
+    this.preferencesController = preferencesController
+    this.addressBookController = addressBookController
+    this.keyringController = keyringController
+    this.provider = provider
+
+    const initState = {
+      threeBoxAddress: null,
+      threeboxSyncing: true,
+      ...opts.initState,
+      syncDone3Box: false,
+    }
+    this.store = new ObservableStore(initState)
+
+    this.init()
+  }
+
+  async init () {
+    const accounts = await this.keyringController.getAccounts()
+    this.address = accounts[0]
+    if (this.address && !(this.box && this.store.getState().syncDone3Box)) {
+      await this.new3Box(this.address)
+    }
+  }
+
+  async _update3Box ({ type }, newState) {
+    const threeBoxSyncing = this.store.getState().threeboxSyncing
+    if (threeBoxSyncing) {
+      await this.box.private.set(type, JSON.stringify(newState))
+    }
+  }
+
+  async new3Box (address) {
+    const threeBoxSyncing = this.store.getState().threeboxSyncing
+    console.log('new3Box threeBoxSyncing', threeBoxSyncing)
+    if (threeBoxSyncing) {
+      this.store.updateState({ syncDone3Box: false })
+      this.address = address
+
+      try {
+        const messageDataToSign = '0x' + Buffer.from('This app wants to view and update your 3Box profile.', 'utf8').toString('hex')
```

Comment:
Can we instead use [`bufferToHex`](https://github.com/ethereumjs/ethereumjs-util/blob/master/docs/README.md#const-buffertohex) or some other util here? That might be simpler to read.